model_compression_toolkit/keras/quantizer/mixed_precision/selective_quantize_config.py

```diff
@@ -0,0 +1,217 @@
+                q.set_active_quantization_config_index(index)
+
+    def set_activation_bit_width_index(self,
+                                       index: int):
+        """
+        Change the "active" bitwidth index the SelectiveQuantizeConfig uses, so
+        a different quantized weight will be used.
+        If attr is passed, only the quantizer that was created for this attribute will be configured.
+        Otherwise, all quantizers the SelectiveQuantizeConfig holds will be configured
+        using the passed index.
+
+        Args:
+            index: Bitwidth index to use.
+            attr: Name of the layer's attribute to configure its corresponding quantizer.
+
+        """
+        if self.enable_activation_quantization:
+            self.activation_selective_quantizer.set_active_quantization_config_index(index)
+
+    def get_weights_and_quantizers(self, layer: Layer) -> List[Tuple[Tensor, Any]]:
+        """
+        Get a list of tuples with weights and the weights quantizers.
+        The layer's attributes are used to get the weights.
+
+        Args:
+            layer: The layer the SelectiveQuantizeConfig is attached to when is wrapped.
+
+        Returns:
+            List of tuples of the layer's weights and the weights quantizers.
+        """
+        return [] if not self.enable_weights_quantization else \
+            [(getattr(layer, self.weight_attrs[i]), self.weight_quantizers[i]) for i in range(len(self.weight_attrs))]
+
+    def get_activations_and_quantizers(self, layer: Layer) -> list:
+        # This QuantizeConfig is for quantizing weights only, so no
+        # implementation is needed for activation quantization.
+        return []
+
+    def set_quantize_weights(self, layer: Layer, quantize_weights: List[Tensor]):
+        """
+        Set the layer weights with new passed weights.
+        Args:
+            layer: Layer to set its attributes.
+            quantize_weights: Quantized weights to set as new weights.
+
+        """
+        if self.enable_weights_quantization:
+            if len(self.weight_attrs) != len(quantize_weights):
+                raise ValueError(
+                    '`set_quantize_weights` called on layer {} with {} '
+                    'weight parameters, but layer expects {} values.'.format(
+                        layer.name, len(quantize_weights), len(self.weight_attrs)))
+
+            for weight_attr, weight in zip(self.weight_attrs, quantize_weights):
+                current_weight = getattr(layer, weight_attr)
+                if current_weight.shape != weight.shape:
+                    raise ValueError('Existing layer weight shape {} is incompatible with'
+                                     'provided weight shape {}'.format(
+                        current_weight.shape, weight.shape))
+
+                setattr(layer, weight_attr, weight)
+
+    def set_quantize_activations(self, layer, quantize_activations: ListWrapper):
+        pass
```

Comment:
Did the model work without using this method?

Reply:
It seems to be working, need to verify after implementing the activation mp search (next PR)